core/capabilities/ccip/ccipsolana/executecodec.go

```diff
@@ -31,6 +31,13 @@ func NewExecutePluginCodecV1(extraDataCodec common.ExtraDataCodec) *ExecutePlugi
 }
 
 func (e *ExecutePluginCodecV1) Encode(ctx context.Context, report cciptypes.ExecutePluginReport) ([]byte, error) {
+	if len(report.ChainReports) == 0 {
+		// OCR3 runs in a constant loop and will produce empty reports, so we need to handle this case
+		// return an empty report, CCIP will discard it on ShouldAcceptAttestedReport/ShouldTransmitAcceptedReport
+		// via validateReport before attempting to decode
```

Comment:
While this is true and this is fine as a sanity check, if this gets called with `ChainReports` being empty there is a bug in cl-ccip somewhere. cc @winder

Reply:
I remember @archseer said previously that he sees ChainReports being empty in e2e testing, and thus had to fix this code.

Reply:
AFAIK there was a bug, and it has been fixed.